packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -335,6 +421,49 @@ class _BottomSheetState extends State<BottomSheet> {
 
 typedef _SizeChangeCallback<Size> = void Function(Size);
 
+class _DragHandle extends StatelessWidget {
+  const _DragHandle({
+    required this.onSemanticsTap,
+    required this.handleHover,
+    required this.materialState,
+  });
+
+  final VoidCallback? onSemanticsTap;
+  final Function(bool) handleHover;
+  final Set<MaterialState> materialState;
+
+  @override
+  Widget build(BuildContext context) {
+    final BottomSheetThemeData bottomSheetTheme = Theme.of(context).bottomSheetTheme;
+    final BottomSheetThemeData m3Defaults = _BottomSheetDefaultsM3(context);
+    final Size dragHandleSize = bottomSheetTheme.dragHandleSize ?? m3Defaults.dragHandleSize!;
+
+    return MouseRegion(
+      onEnter: (PointerEnterEvent event) => handleHover(true),
+      onExit: (PointerExitEvent event) => handleHover(false),
+      child: Semantics(
+        label: MaterialLocalizations.of(context).modalBarrierDismissLabel,
+        container: true,
+        onTap: onSemanticsTap,
+        child: SizedBox(
+          height: kMinInteractiveDimension,
+          width: kMinInteractiveDimension,
+          child: Center(
+            child: Container(
+              height: dragHandleSize.height,
+              width: dragHandleSize.width,
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.circular(dragHandleSize.height/2),
+                color: MaterialStateProperty.resolveAs<Color?>(bottomSheetTheme.dragHandleColor, materialState) ?? m3Defaults.dragHandleColor!,
```

Comment:
To resolve the effective drag handle color here - assuming that the widget's dragHandleColor was passed along as a _DragHandle parameter:
```dart
  MaterialStateProperty.resolveAs<Color?>(dragHandleColor) ??
  MaterialStateProperty.resolveAs<Color?>(bottomSheetTheme..dragHandleColor) ??
  m3Defaults.dragHandleColor;
```